dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/datafile/ingest/IngestReport.java

```diff
@@ -121,6 +130,30 @@ public void setEndTime(Date endTime) {
         this.endTime = endTime;
     }
 
+    public List<String> getReportArguments() {
+        return reportArguments;
+    }
+
+    // -------------------- LOGIC --------------------
+
+    public String getIngestReportMessage() {
+        IngestError errorKey = getErrorKey();
+        if (errorKey != null) {
+
+            if (!getReportArguments().isEmpty()) {
+                String errorMessage = IngestError.getErrorMessage(errorKey, getReportArguments());
+
+                return errorMessage.isEmpty() ? IngestError.getDefaultErrorMessage() : errorMessage;
+            }
+
+            String errorMessage = IngestError.getErrorMessage(errorKey);
+
+            return errorMessage.isEmpty() ? IngestError.getDefaultErrorMessage() : errorMessage;
+        }
+
+        return IngestError.getDefaultErrorMessage();
```

Comment:
This looks like a great use case for `Optional` and Null Object Pattern. I'd see this whole method simply like this:
```
return Optional.ofNullable(errorKey)
        .orElse(IngestError.UNKNOWN_ERROR)
        .getMessage(Optional.ofNullable(errorArguments)
                .orElseGet(Collections::emptyList));
```
And `getMessage(..)` should handle empty arguments and never return empty `String`.